### PR TITLE
[5.8] added throws to comments

### PR DESCRIPTION
**Pull request by hassanamjad1129**

### CODE REVIEW

File: src/Illuminate/Bus/Dispatcher.php [hunk 1]
```diff
@@ -112,8 +112,9 @@ public function hasCommandHandler($command)
     /**
      * Retrieve the handler for a command.
      *
-     * @param  mixed  $command
+     * @param  mixed $command
```
Comment (AbdelElrafa) [inline src/Illuminate/Bus/Dispatcher.php]: You're going to need to revert all these param space changes. There should be 2 spaces before and after the param type.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -149,9 +149,10 @@ public function replace($path, $content)
     /**
      * Prepend to a file.
      *
-     * @param  string  $path
-     * @param  string  $data
+     * @param  string $path
+     * @param  string $data
```
Comment (AbdelElrafa) [inline src/Illuminate/Filesystem/Filesystem.php]: You're going to need to revert all these param space changes. There should be 2 spaces before and after the param type.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 3]
```diff
@@ -149,9 +149,10 @@ public function replace($path, $content)
...
      * @return int
+     * @throws FileNotFoundException
```
Comment (AbdelElrafa) [inline src/Illuminate/Filesystem/Filesystem.php]: You need to add the fully qualified name of the exception. `\Illuminate\Contracts\Filesystem\FileNotFoundException`

File: src/Illuminate/Filesystem/FilesystemAdapter.php [hunk 4]
```diff
@@ -265,10 +265,11 @@ public function setVisibility($path, $visibility)
...
      * Prepend to a file.
      *
-     * @param  string  $path
-     * @param  string  $data
-     * @param  string  $separator
+     * @param  string $path
+     * @param  string $data
+     * @param  string $separator
      * @return bool
+     * @throws ContractFileNotFoundException
```
Comment (AbdelElrafa) [inline src/Illuminate/Filesystem/FilesystemAdapter.php]: This isn't the name of the Exception. The exception is `\Illuminate\Contracts\Filesystem\FileNotFoundException`

File: src/Illuminate/Filesystem/FilesystemAdapter.php [hunk 5]
```diff
@@ -282,10 +283,11 @@ public function prepend($path, $data, $separator = PHP_EOL)
...
      * Append to a file.
      *
-     * @param  string  $path
-     * @param  string  $data
-     * @param  string  $separator
+     * @param  string $path
+     * @param  string $data
+     * @param  string $separator
      * @return bool
+     * @throws ContractFileNotFoundException
```
Comment (AbdelElrafa) [inline src/Illuminate/Filesystem/FilesystemAdapter.php]: This isn't the name of the Exception. The exception is `\Illuminate\Contracts\Filesystem\FileNotFoundException`

File: src/Illuminate/Filesystem/FilesystemAdapter.php [hunk 6]
```diff
@@ -282,10 +283,11 @@ public function prepend($path, $data, $separator = PHP_EOL)
     /**
      * Append to a file.
      *
-     * @param  string  $path
-     * @param  string  $data
-     * @param  string  $separator
+     * @param  string $path
+     * @param  string $data
+     * @param  string $separator
```
Comment (AbdelElrafa) [inline src/Illuminate/Filesystem/FilesystemAdapter.php]: You're going to need to revert all these param space changes. There should be 2 spaces before and after the param type.

**Comment by ahinkle** [comment]
Please don't remove the spaces after the attributes. Refer to the [Contribution Guide](https://laravel.com/docs/5.8/contributions#coding-style).

**Comment by browner12** [comment]
Laravel seems to currently avoid the `@throws` docblocks. If we decide to do it here, I believe there are **a lot** of other places for them as well.

**Comment by mfn** [comment]
There's currently no real style here, it's hit or miss.

Something similar came up just a few days and I collected my thoughts here already: https://github.com/laravel/ideas/issues/1573#issuecomment-475894956

**Comment by driesvints** [comment]
I agree with @mfn. I think it's best to only document `@throw` annotations on the methods that actually throw the exception. You can ignore the warnings in your IDE if you like.

**Comment by taylorotwell** [comment]
Has a lot of unrelated formatting changes.

**Pull request closed without merging**